clock.go

```diff
@@ -2,9 +2,9 @@ package abc
 
 import "time"
 
-// Clock is an interface created to be able to
+// clockT is an interface created to be able to
 // mock time.Now() easily.
-type Clock interface {
+type clockT interface {
```

Comment:
Why `clockT` instead of `clock`?

Reply:
because otherwise you will have variable 'clock' in some struct of the same name of the type 'clock' and I think it's confusing... For example see

https://github.com/TimSatke/abc/blob/develop/named_logger.go#L23

IMHO it's better to keep apart variable names and type names (hence the 'T' for Type)

Reply:
Maybe change `clock clock` to `cl clock`?
As [Patrick Stephens answer](https://gophers.slack.com/archives/C029RQSEE/p1565273888087800) mentioned, `clock clock` has a potential ambiguity as it could be confused with embedding upon use.